aten/src/ATen/native/nested/cuda/NestedTensorTransformerFunctions.cu

```diff
@@ -556,32 +556,32 @@ inline std::tuple<dim3, dim3, StackArray<int64_t>> check_shape_and_partition_(
     const Tensor& values,
     const std::vector<Tensor>& offsets,
     const Tensor& dense_tensor) {
-  const int outer_dense_size = dense_tensor.size(0);
+  const auto outer_dense_size = dense_tensor.size(0);
   TORCH_CHECK(
       outer_dense_size == offsets[0].numel() - 1,
       "outer_dense_size, ",
       outer_dense_size,
       " != offsets[0].numel() - 1, ",
       offsets[0].numel() - 1);
-  const int inner_dense_size = dense_tensor.size(-1);
+  const auto inner_dense_size = dense_tensor.size(-1);
   TORCH_CHECK(
       inner_dense_size == values.size(-1),
       "inner_dense_size, ",
       inner_dense_size,
       " != values.size(-1), ",
       values.size(-1));
-  const int jagged_folded_size =
+  const auto jagged_folded_size =
       dense_tensor.numel() / (outer_dense_size * inner_dense_size);
 
-  const int threads_x =
+  const int64_t threads_x =
```

Comment:
We shouldn't change the semantic from using 32bit to 64bit in places like this.

We should also ensure the int -> auto changes don't switch 32bit ints to 64bit.